pkg/controller/master/upgrade/upgrade_controller.go

```diff
@@ -256,38 +257,63 @@ func (h *upgradeHandler) OnRemove(_ string, upgrade *harvesterv1.Upgrade) (*harv
 	}
 
 	logrus.Debugf("Deleting upgrade %s", upgrade.Name)
+	return upgrade, h.cleanup(upgrade, true)
+}
+
+func (h *upgradeHandler) cleanup(upgrade *harvesterv1.Upgrade, cleanJobs bool) error {
+	// delete vm and images
+	repo := NewUpgradeRepo(h.ctx, upgrade, h)
+	if err := repo.deleteVM(); err != nil {
+		return err
+	}
+
+	// remove rkeConfig in fleet-local/local cluster
+	cluster, err := h.clusterCache.Get("fleet-local", "local")
+	if err != nil {
+		return err
+	}
+	clusterToUpdate := cluster.DeepCopy()
+	clusterToUpdate.Spec.RKEConfig = nil
```

Comment:
```suggestion
	clusterToUpdate.Spec.RKEConfig = &provisioningv1.RKEConfig{}
```

assigning nil here seems to delete the cluster.